jobclient/java/src/main/java/com/twosigma/cook/jobclient/Job.java

```diff
@@ -722,6 +724,165 @@ public Builder setDatasets(JSONArray datasets) {
             _datasets = datasets;
             return this;
         }
+
+
+        /**
+         * Parse a JSON object into this Builder object, e.g.
+         * <p>
+         * <pre>
+         * <code>
+         * {
+         *    "status" : "completed",
+         *    "mem" : 1000,
+         *    "uuid" : "26719da8-394f-44f9-9e6d-8a17500f5109",
+         *    "cpus" : 1.5,
+         *    "command" : "echo hello world",
+         * }
+         * </code>
+         * </pre>
+         *
+         * @param json {@link JSONObject} specifies a JSONObject with Builder parameters.
+         * @param decorator  specifies an instance decorator expected to decorate instances parsed from JSON string.
+         *                   If it is null, it will do nothing with it.
+         * @return {@link Builder}.
+         * @throws JSONException
+         */
+        public Builder parseFromJSON(JSONObject json, InstanceDecorator decorator)
```

Comment:
Basically this function doesn't enforce that any specific field is set, if any of them are set, the Builder will do the right thing and set the right field. Fields are only enforced on `build()` (e.g. if cpus is not defined, that will be caught by `build()`).